packages/smart-contracts/scripts-create2/contract-setup/setups.ts

```diff
@@ -0,0 +1,168 @@
+import { HardhatRuntimeEnvironmentExtended } from '../types';
+import { erc20SwapConversionArtifact } from '../../src/lib';
+import { batchPaymentsArtifact } from '../../src/lib';
+import utils from '@requestnetwork/utils';
+import {
+  updateBatchPaymentFees,
+  updatePaymentErc20FeeProxy,
+  updatePaymentEthFeeProxy,
+  updateChainlinkConversionPath,
+  updateRequestSwapFees,
+  updateSwapRouter,
+} from './adminTasks';
+
+/**
+ * Updates the values of the batch fees of the BatchPayments contract, if needed
+ * @param contractAddress address of the BatchPayments Proxy
+ * @param hre Hardhat runtime environment
+ */
+const setupBatchPayments = async (
+  contractAddress: string,
+  hre: HardhatRuntimeEnvironmentExtended,
+): Promise<void> => {
+  // Setup contract parameters
+  const batchPaymentContract = new hre.ethers.Contract(
+    contractAddress,
+    batchPaymentsArtifact.getContractAbi(),
+  );
+  await Promise.all(
+    hre.config.xdeploy.networks.map(async (network) => {
+      let provider;
+      if (network === 'celo') {
+        provider = utils.getCeloProvider();
+      } else {
+        provider = utils.getDefaultProvider(network);
+      }
+      const wallet = new hre.ethers.Wallet(hre.config.xdeploy.signer, provider);
+      const signer = wallet.connect(provider);
+      const batchPaymentConnected = await batchPaymentContract.connect(signer);
+      const adminNonce = await signer.getTransactionCount();
+      const gasPrice = await provider.getGasPrice();
+
+      // start from the adminNonce, increase gasPrice if needed
+      await Promise.all([
+        updateBatchPaymentFees(batchPaymentConnected, adminNonce, gasPrice.mul(2)),
+        updatePaymentErc20FeeProxy(batchPaymentConnected, network, adminNonce + 1, gasPrice.mul(2)),
+        updatePaymentEthFeeProxy(batchPaymentConnected, network, adminNonce + 2, gasPrice.mul(2)),
+      ]);
+    }),
+  );
+  console.log('Setup for setupBatchPayment successful');
+};
```

Comment:
@olivier7delf The batch setup was moved here, I copied your latest changes for multiplying the gasPrice by 2 as well.

Reply:
I saw it, thanks.
(without gas mul, it failed to deploy...)